node/pkg/fetcher/fetcher.go

```diff
@@ -1 +1,153 @@
 package fetcher
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"time"
+
+	"bisonai.com/orakl/node/pkg/bus"
+	"bisonai.com/orakl/node/pkg/db"
+	"bisonai.com/orakl/node/pkg/utils"
+)
+
+const FETCHER_FREQUENCY = 2 * time.Second
+
+func New(bus *bus.MessageBus) *Fetcher {
+	return &Fetcher{
+		Adapters: make([]AdapterDetail, 0),
+		Bus:      bus,
+	}
+}
+
+func (f *Fetcher) Run(ctx context.Context) error {
+	err := f.initialize(ctx)
+	if err != nil {
+		return err
+	}
+
+	ticker := time.NewTicker(FETCHER_FREQUENCY)
+	go func() {
+		for range ticker.C {
+			err := f.fetchAll(ctx)
+			if err != nil {
+				fmt.Println(err)
+			}
+		}
+	}()
+
+	return nil
+}
+
+func (f *Fetcher) fetchAll(ctx context.Context) error {
+	iterationDuration := FETCHER_FREQUENCY / time.Duration(len(f.Adapters))
+	for _, adapter := range f.Adapters {
+		start := time.Now()
+
+		result, err := f.fetch(adapter)
```

Comment:
How about using `results` for array variables, follow naming convention of this whole file